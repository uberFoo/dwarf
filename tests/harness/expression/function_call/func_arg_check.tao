fn test(a: int, b: float, c: string) -> int {
    a + b as int + c.len()
}

fn main() {
    print(test(1, 2, "three"));
}
