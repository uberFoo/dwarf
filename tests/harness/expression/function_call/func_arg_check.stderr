Interpreter exited with:
[31mError:[0m Type mismatch: expected `[3;35mfloat[0m`, found `[3;35mint[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0mfunc_arg_check:6:19[38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m6 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249mp[0m[38;5;249mr[0m[38;5;249mi[0m[38;5;249mn[0m[38;5;249mt[0m[38;5;249m([0m[38;5;249mt[0m[38;5;249me[0m[38;5;249ms[0m[38;5;249mt[0m[38;5;249m([0m[38;5;249m1[0m[38;5;249m,[0m[38;5;249m [0m[33m2[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249m"[0m[38;5;249mt[0m[38;5;249mh[0m[38;5;249mr[0m[38;5;249me[0m[38;5;249me[0m[38;5;249m"[0m[38;5;249m)[0m[38;5;249m)[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m                   [33mâ”¬[0m  
 [38;5;240m  â”‚[0m                   [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m expected [33m[3;35mfloat[0m[0m
 [38;5;240m  â”‚[0m                   [31mâ”‚[0m  
 [38;5;240m  â”‚[0m                   [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m found [33m[3;35mint[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

